packages/next/taskfile.js

```diff
@@ -1672,14 +1670,14 @@ export async function ncc_send(task, opts) {
     .target('compiled/send')
 }
 // eslint-disable-next-line camelcase
-// NB: Used by other dependencies, but Vercel version is a duplicate
-// version so can be inlined anyway (although may change in future)
-externals['source-map'] = 'next/dist/compiled/source-map'
+externals['source-map'] = 'next/dist/compiled/@jridgewell/source-map'
```

Comment:
Is `@jridgewell/source-map` a drop-in replacement of `source-map`? If it is not, it is better not to use `externals['source-map']` here.

```suggestion
externals['@jridgewell/source-map'] = 'next/dist/compiled/@jridgewell/source-map'
```

Reply:
This is actually intentional. It's a drop in replacement for the core API, which all that's needed in this case. There are unimplemented APIs, but they're not necessary.

Reply:
There are way too many packages that require the `source-map`. Will `@jridgewell/source-map` be compatible with all of them?

```
$ pnpm why source-map --filter=next
Legend: production dependency, optional only, dev only

next@12.3.2-canary.32 /Users/sukka/Project/next.js/packages/next

dependencies:
node-sass 7.0.3
└─┬ sass-graph 4.0.1
  └─┬ scss-tokenizer 0.4.3
    └── source-map 0.7.3

devDependencies:
@ampproject/toolbox-optimizer 2.8.3
├─┬ cssnano-simple 2.0.0
│ ├─┬ cssnano-preset-simple 2.0.0
│ │ └─┬ postcss 8.2.13 peer
│ │   └── source-map 0.6.1
│ └─┬ postcss 8.2.13 peer
│   └── source-map 0.6.1
├─┬ postcss 8.2.13
│ └── source-map 0.6.1
└─┬ terser 5.5.1
  ├── source-map 0.7.3
  └─┬ source-map-support 0.5.20
    └── source-map 0.6.1
@taskr/watch 1.1.0
└─┬ chokidar 1.7.0
  └─┬ readdirp 2.2.1
    └─┬ micromatch 3.1.10
      ├─┬ braces 2.3.2
      │ └─┬ snapdragon 0.8.2
      │   └── source-map 0.5.7
      ├─┬ extglob 2.0.4
      │ ├─┬ expand-brackets 2.1.4
      │ │ └─┬ snapdragon 0.8.2
      │ │   └── source-map 0.5.7
      │ └─┬ snapdragon 0.8.2
      │   └── source-map 0.5.7
      ├─┬ nanomatch 1.2.13
      │ └─┬ snapdragon 0.8.2
      │   └── source-map 0.5.7
      └─┬ snapdragon 0.8.2
        └── source-map 0.5.7
@types/webpack-sources 0.1.5
└── source-map 0.6.1
mini-css-extract-plugin 2.4.3
└─┬ webpack 5.74.0 peer
  └─┬ terser-webpack-plugin 5.2.4
    ├── source-map 0.6.1
    └─┬ terser 5.14.1
      └─┬ source-map-support 0.5.20
        └── source-map 0.6.1
react-server-dom-webpack 0.0.0-experimental-cb5084d1c-20220924
└─┬ webpack 5.74.0 peer
  └─┬ terser-webpack-plugin 5.2.4
    ├── source-map 0.6.1
    └─┬ terser 5.14.1
      └─┬ source-map-support 0.5.20
        └── source-map 0.6.1
sass-loader 12.4.0
├─┬ node-sass 7.0.3 peer
│ └─┬ sass-graph 4.0.1
│   └─┬ scss-tokenizer 0.4.3
│     └── source-map 0.7.3
└─┬ webpack 5.74.0 peer
  └─┬ terser-webpack-plugin 5.2.4
    ├── source-map 0.6.1
    └─┬ terser 5.14.1
      └─┬ source-map-support 0.5.20
        └── source-map 0.6.1
source-map 0.6.1
terser 5.14.1
└─┬ source-map-support 0.5.20
  └── source-map 0.6.1
webpack 5.74.0
└─┬ terser-webpack-plugin 5.2.4
  ├── source-map 0.6.1
  └─┬ terser 5.14.1
    └─┬ source-map-support 0.5.20
      └── source-map 0.6.1
webpack-sources 1.4.3
└── source-map 0.6.1
```